code/greedy_algorithms/src/huffman_coding/huffman_coding_encoding.java

```diff
@@ -0,0 +1,201 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.PriorityQueue;
+import java.util.Scanner;
+import java.util.TreeMap;
+
+/* Huffman coding , decoding */
+class Huffman {
+    static final boolean readFromFile = false;
+    static final boolean newTextBasedOnOldOne = false;
+
+    static PriorityQueue<Node> nodes = new PriorityQueue<>((o1, o2) -> (o1.value < o2.value) ? -1 : 1);
+    static TreeMap<Character, String> codes = new TreeMap<>();
+    static String text = "";
+    static String encoded = "";
+    static String decoded = "";
+    static int ASCII[] = new int[128];
+
+    public static void main(String[] args) throws FileNotFoundException {
+        Scanner scanner = (readFromFile) ? new Scanner(new File("input.txt")) : new Scanner(System.in);
+        int decision = 1;
+        while (decision != -1) {
```

Comment:
While this code is amazing and all, I would like to make a suggestion. Currently your Huffman class does console output as well as encoding/decoding. I suggest that you leave all console output/menu output to main, and let the functions of class Huffman return a string/value for their operations. This will make the code cleaner and well organized. Then you can move all console related functions to their own class, and only keep Huffman operations in this class.